src/test/java/springsamurais/toyswapbackend/controller/ListingControllerTest.java

```diff
@@ -90,5 +90,42 @@ void testGetListingByIdWhenListingDoesNotExistThenReturnNotFound() throws Except
     }
 
 
+    // Delete Testings
+    @Test
+    void testDeleteListing_Success() throws Exception {
+        Long listingID = 1L;
+
+        mockMvc.perform(MockMvcRequestBuilders.delete("/api/v1/listings/{listingID}", listingID))
+                .andExpect(MockMvcResultMatchers.status().isNoContent());
+
+        verify(mockListingService).deleteListingById(listingID);
+    }
+
+    @Test
+    void testDeleteListing_NotFound() throws Exception {
+        Long listingID = 1L;
+        doThrow(new ListingNotFoundException("Listing not found")).when(mockListingService).deleteListingById(listingID);
+
+        mockMvc.perform(MockMvcRequestBuilders.delete("/api/v1/listings/{listingID}", listingID))
+                .andExpect(MockMvcResultMatchers.status().isNotFound());
+    }
 
+    @Test
+    void testDeleteListingsByMember_Success() throws Exception {
+        Long memberID = 1L;
+
+        mockMvc.perform(MockMvcRequestBuilders.delete("/api/v1/listings/member/{memberID}", memberID))
+                .andExpect(MockMvcResultMatchers.status().isNoContent());
+
+        verify(mockListingService).deleteListingsByMember(memberID);
+    }
+
+    @Test
+    void testDeleteListingsByMember_NotFound() throws Exception {
+        Long memberID = 1L;
+        doThrow(new ListingNotFoundException("No listings found for member")).when(mockListingService).deleteListingsByMember(memberID);
+
+        mockMvc.perform(MockMvcRequestBuilders.delete("/api/v1/listings/member/{memberID}", memberID))
+                .andExpect(MockMvcResultMatchers.status().isNotFound());
+    }
```

Comment:
all good. I would start changing the expected HTTP codes and refactor.

Reply:
HTTP codes Changed and refactored